src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -285,32 +285,60 @@ def _publish_trade_event(self, trade, is_bid_trade):
             # the bid clearing and one for the offer clearing.
             return
 
-        event_response_dict = {"device_info": self._device_info_dict,
-                               "event": "trade",
-                               "trade_id": trade.id,
-                               "time": trade.time.isoformat(),
-                               "price": trade.offer.price,
-                               "energy": trade.offer.energy,
-                               "fee_price": trade.fee_price,
-                               "area_uuid": self.device.uuid,
-                               "seller": trade.seller
-                               if trade.seller == self.device.name else "anonymous",
-                               "buyer": trade.buyer
-                               if trade.buyer == self.device.name else "anonymous",
-                               "residual_id": trade.residual.id
-                               if trade.residual is not None else "None"}
-
-        bid_offer_key = 'bid_id' if is_bid_trade else 'offer_id'
-        event_response_dict["event_type"] = "buy" \
-            if trade.buyer == self.device.name else "sell"
-        event_response_dict[bid_offer_key] = trade.offer.id
-
         if self.is_aggregator_controlled:
+            event_response_dict = {'event': 'trade',
+                                   'asset_id': self.device.uuid,
+                                   'trade_id': trade.id,
+                                   'time': trade.time.isoformat(),
+                                   'trade_price': trade.offer.price,
+                                   'traded_energy': trade.offer.energy,
+                                   'total_fee': trade.fee_price,
+                                   'local_market_fee':
+                                       self.area.current_market.fee_class.grid_fee_rate
+                                       if self.area.current_market is not None else "None",
+                                   'attributes': {},
+                                   'seller': trade.seller
+                                   if trade.seller_id == self.device.uuid else 'anonymous',
+                                   'buyer': trade.buyer
+                                   if trade.buyer_id == self.device.uuid else 'anonymous',
```

Comment:
`seller_id` & `buyer_id ` should always be equal to `device_uuid` for offers/bids of strategies. Have you already encountered where they were not equal?

Reply:
If I am not mistaken trade seller and buyer are not always equal to the device uuid. Seller/buyer can either be another device on the same market, or the market itself, according to the usecase.

Reply:
> `seller_id` & `buyer_id ` should always be equal to `device_uuid` for offers/bids of strategies. Have you already encountered where they were not equal?

Sorry, dont get your question. Do you mean why we return "anonymous" for if the device is not the seller or buyer? I really can not remember. I guess that this was due to a privacy issue.

Reply:
> If I am not mistaken trade seller and buyer are not always equal to the device uuid. Seller/buyer can either be another device on the same market, or the market itself, according to the usecase.

Ah yes, but why is it not good to return it in any case? I understand that the buyer_origin should be hidden, but in the same market, it should be ok to expose the seller or buyer right?

Reply:
Depends on the usecase, for houses this is definitely ok. For the higher levels in the hierarchy though, this might be a security concern (for instance, if a power plant sells energy to an area). At least that was the original though behind that, when we were much stricter about privacy than we unfortunately are now.

Reply:
> Sorry, dont get your question. Do you mean why we return "anonymous" for if the device is not the seller or buyer? I really can not remember. I guess that this was due to a privacy issue.

Earlier, I skipped the case where strategies market having external traded bids/offers BUT still always using `anonymous` for external agent may not be needed. For `buyer_origin/seller_origin` may be okay to mask it as anonymous but at least for `buyer/seller` as they show only the immediate connected market masking them as `anonymous` could be skipped, anyways will leave it for future feedback of users